### PR TITLE
Isolating the git user and email into a UserSettingsView

### PR DESCRIPTION
**Pull request by StanleyGoldman**
Part of #251
#261 depends on this

Depends on:
- [x] #273 
- [x] #277 

- There are no visible changes to the UI as a result of this change
- These changes also prevent the save of a blank name or email

### CODE REVIEW

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PopupWindow.cs [hunk 1]
```diff
@@ -93,6 +93,11 @@ public override void OnSelectionChange()
             ActiveView.OnSelectionChange();
         }
 
+        public override bool IsBusy
+        {
+            get { return ActiveView.IsBusy; }
```
Comment (shana) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PopupWindow.cs]: This change is probably a glitchy merge, the property should be at the bottom

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/SettingsView.cs [hunk 2]
```diff
@@ -68,18 +69,25 @@ public override void OnEnable()
...
         {
             base.OnDisable();
+            userSettingsView.OnDisable();
             DetachHandlers(Repository);
         }
 
         public override void OnDataUpdate()
         {
             base.OnDataUpdate();
+            if (userSettingsView != null)
```
Comment (shana) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/SettingsView.cs]: When is this ever null?

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/SettingsView.cs [hunk 3]
```diff
@@ -89,9 +97,15 @@ public override void OnRepositoryChanged(IRepository oldRepository)
             Refresh();
         }
 
+        public override bool IsBusy
+        {
+            get { return isBusy || userSettingsView.IsBusy; }
```
Comment (shana) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/SettingsView.cs]: Bottom of file

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/SettingsView.cs (hunk 2 above)
Comment (StanleyGoldman) [reply src/UnityExtension/Assets/Editor/GitHub.Unity/UI/SettingsView.cs]: You are right, removing the null check

**Pull request merged**